src/v/utils/retry_chain_node.h

```diff
@@ -371,11 +372,13 @@ class retry_chain_node {
     /// Initial backoff value
     milliseconds_uint16_t _backoff;
     /// Deadline for retry attempts
-    ss::lowres_clock::time_point _deadline;
+    time_point _deadline;
     /// optional parent node or (if root) abort source for all fibers
-    std::variant<retry_chain_node*, ss::abort_source*> _parent;
+    std::variant<basic_retry_chain_node*, ss::abort_source*> _parent;
 };
 
+using retry_chain_node = basic_retry_chain_node<ss::lowres_clock>;
```

Comment:
this is a drop-in replacement for existing code

Reply:
question: how do you decide between basic_T vs T_base? Where this naming convention is from?

Reply:
Because it's not a base class, it's similar to `std::basic_string` situation where `std::string` is just a `typedef basic_string<char> string`.